magicctapipe/scripts/lst1_magic/coincident_brokenGPS.py

```diff
@@ -0,0 +1,418 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+"""
+This script search the time offset for using MAGIC data which the GPS is broken.
+The offset time in such a case is large (few second) and changed gradually (few us/sec). The drift speed is known as constant in each subrun.
+In this script we use the subrun files of MAGIC and LST as input files, and make a combination for each event timestamp. From each combination, the best matched time will be found.
+It will take few tens of minutes to make each npy files. Basically, each MAGIC run contains ~10 subruns and so this step take few hours to be finished.
+
+Usage:
+$ python lst1_magic_event_coincidence.py data
+
+If you have some runs the input data directory should be separated for each run and use them to save the time, as like,
+data1/LST/
+dl1_LST-1.Run17220.0000.h5 dl1_LST-1.Run17220.0001.h5 ...
+data1/MAGIC/
+dl1_MAGIC.Run05114133.001.h5 dl1_MAGIC.Run05114133.002.h5 ...
+
+data2/LST/
+dl1_LST-1.Run17221.0000.h5 dl1_LST-1.Run17221.0001.h5 ...
+data2/MAGIC/
+dl1_MAGIC.Run05114134.001.h5 dl1_MAGIC.Run05114134.002.h5 ...
+
+$ python lst1_magic_event_coincidence.py data1
+In parallel,
+$ python lst1_magic_event_coincidence.py data2
+...
+"""
+
+import glob
+import os
+import subprocess
+import sys
+
+import numpy as np
+import pandas as pd
+
+from magicctapipe.io import find_offset
+
+file_dl1_dir = sys.argv[1]
+outdir = "time_offset"
+config = "../../config_dyn.yaml"
+
+magic_run = int(
+    pd.read_hdf(
+        glob.glob(file_dl1_dir + "/MAGIC/dl1_MAGIC.*Run*h5")[0], key="events/parameters"
+    )["obs_id"].mean()
```

Comment:
the code will not work fine if you put together a few runs, maybe better to check if there is only one unique obs_id in the sample and if no return an error, same for LST below